app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/DefaultTasksRepository.kt

```diff
@@ -95,68 +83,51 @@ class DefaultTasksRepository(
         }
     }
 
-    override fun getTaskStream(taskId: String): Flow<Task?> {
-        return tasksDao.observeTaskById(taskId).map { it.toExternalModel() }
+    private suspend fun sendTasksToNetworkDataSource() {
+        // Real apps may want to use a proper sync strategy here to avoid data conflicts.
+        val localTasks = tasksDao.getTasks()
+        tasksNetworkDataSource.saveTasks(localTasks.toNetworkModels())
```

Comment:
Same comment applies to all places where we do mapping.

We don't know how expensive retrieving from the local data source and mapping the models is. We should move the execution to the `defaultDispatcher`. We could be talking about hundreds / thousands of tasks and that could block the calling thread.

Reply:
Good point. I've moved execution to `Dispatchers.Default` for all *bulk* network, local db and mapping operations. Do you think it's worth doing this for single operations as well, like `getTask`? To me that seems like overkill.

Reply:
Just to note that only bulk network and mapping operations need `withContext`, Room will take care of using the correct dispatcher for local database operations.